Source/Dafny/Verifier/Translator.BoogieFactory.cs

```diff
@@ -867,5 +867,24 @@ enum BuiltinFunction {
       fvars.Add(BplFormalVar(name, ty, incoming, out var e));
       return e;
     }
+
+    public static IToken ToDafnyToken(Bpl.IToken exprTok) {
+      if (exprTok == null) {
```

Comment:
Can't we just cast to `Dafny.Token` since those are the only tokens that we are passing to Boogie? Or does Boogie generate fresh `Boogie.IToken` instances?